File: 3043. Find the Length of the Longest Common Prefix/solution.py
class Solution:
    def longestCommonPrefix(self, arr1: List[int], arr2: List[int]) -> int:
        st = set()
        n1, n2 = len(arr1), len(arr2)
        mx = 0

        for num in arr1:
            curr = ""
            for x in str(num):
                curr += x
                st.add(curr)

        for num in arr2:
            curr = ""
            for x in str(num):
                curr += x
                if curr in st:
                    mx = max(mx, len(curr))

        return mx

File: 3043. Find the Length of the Longest Common Prefix/solution.java
class Solution {
    public int longestCommonPrefix(int[] arr1, int[] arr2) {

        HashSet<String> hs = new HashSet<>();
        int n1 = arr1.length, n2 = arr2.length, mx = 0;

       
        for (int i = 0; i < n1; i++) {
            String curr = "";
            String str = Integer.toString(arr1[i]);
            for (char x : str.toCharArray()) {
                curr += x;
                hs.add(curr);
            }
        }

        
        for (int i = 0; i < n2; i++) {
            String curr = "";
            String str = Integer.toString(arr2[i]);
            for (char x : str.toCharArray()) {
                curr += x;
                if (hs.contains(curr)) {
                    mx = Math.max(mx, curr.length());
                }
            }
        }

        return mx;
    }
}